infra/main.bicep

```diff
@@ -0,0 +1,314 @@
+
+/* Spring Boot Telemetry */
+
+@maxLength(32)
+@description('Name of the SpringBoot Telemetry Container App to deploy. If not specified, a name will be generated. The maximum length is 32 characters.')
+param springBootContainerAppName string = ''
+
+@description('Set if the SpringBoot Telemetry Container aAp already exists.')
+param springBootAppExists bool = false
+
+/******************************************************************************/
+/*                                 VARIABLES                                  */
+/******************************************************************************/
+
+var abbreviations = loadJsonContent('./abbreviations.json')
+
+// tags that should be applied to all resources.
+var tags = {
+  // Tag all resources with the environment name.
+  'azd-env-name': environmentName
+}
+
+/******************************* Resource Names *******************************/
+
+
+// Generate a unique token to be used in naming resources.
+// Remove linter suppression after using.
+#disable-next-line no-unused-vars
+var resourceToken = toLower(uniqueString(subscription().id, environmentName, location))
+
+@description('Name of the environment with only alphanumeric characters. Used for resource names that require alphanumeric characters only.')
+var alphaNumericEnvironmentName = replace(replace(environmentName, '-', ''), ' ', '')
+
+var _containerAppsEnvironmentName = !empty(containerAppsEnvironmentName) ? containerAppsEnvironmentName : take('${abbreviations.appManagedEnvironments}${environmentName}', 60)
+var _containerRegistryName = !empty(containerRegistryName) ? containerRegistryName : take('${abbreviations.containerRegistryRegistries}${take(alphaNumericEnvironmentName, 35)}${resourceToken}', 50)
+var _logAnalyticsName = !empty(logAnalyticsWorkspaceName) ? logAnalyticsWorkspaceName : take('${abbreviations.operationalInsightsWorkspaces}${environmentName}', 63)
+var _applicationInsightsName = !empty(applicationInsightsName) ? applicationInsightsName : take('${abbreviations.insightsComponents}${environmentName}', 255)
+var _applicationInsightsDashboardName = !empty(applicationInsightsDashboardName) ? applicationInsightsDashboardName : take('${abbreviations.portalDashboards}${environmentName}', 160)
+var _quarkusContainerAppName = !empty(quarkusContainerAppName) ? quarkusContainerAppName : take('${abbreviations.appContainerApps}quarkus-${environmentName}', 32)
+var _springBootContainerAppName = !empty(springBootContainerAppName) ? springBootContainerAppName : take('${abbreviations.appContainerApps}spring-boot-${environmentName}', 32)
+var _postgresFlexibleServerName = !empty(postgresFlexibleServerName) ? postgresFlexibleServerName : take(toLower('${abbreviations.dBforPostgreSQLServers}${take(environmentName, 44)}-${resourceToken}'), 63)
+var _keyVaultName = !empty(keyVaultName) ? keyVaultName : take('${abbreviations.keyVaultVaults}${take(alphaNumericEnvironmentName, 8)}${resourceToken}', 24)
```

Comment:
I think it is a better practice to provide these names only in the one line where you need them. If you store them as variables, you may be tempted to reference the variable name instead of module.outputs.name, and then Bicep won't realize there's a dependency between two modules.

Reply:
I understand your concern. The generation of names and counting of how many characters are truncated is quite complex and prone to mistakes. I prefer to have all name wrangling in the same place for the main file. Also, the extra long lines make for less readable resources & modules.
Misuse of those variables is mitigated by their '_' prefix.